lib/ClangImporter/ImportDecl.cpp

```diff
@@ -4830,8 +4832,9 @@ namespace {
                                               getVersion()});
         if (known != Impl.ImportedDecls.end()) {
```

Comment:
If we already have a known nullptr, maybe it should just return nullptr?

Reply:
This one actually changes the behavior pretty significantly. Lot of tests fail. Do we want it to stop here and return nullptr instead of continuing with all the code that happens after `ImportedName importedName;` or 
`// Determine whether the function is throwing and/or async.` (for the next one you commented on) ?

Reply:
> Oh also, the logic of this code was to return the decl if `isAcceptableResult` returns true. If the decl is nullptr due to a bad import attempt can we exactly consider this nullptr as a meeting these criteria:
> 
> ```
>     static bool isAcceptableResult(Decl *fn,
>                                    Optional<AccessorInfo> accessorInfo) {
>       // We can't safely re-use the same declaration if it disagrees
>       // in accessor-ness.
>       auto accessor = dyn_cast<AccessorDecl>(fn);
>       if (!accessorInfo)
>         return accessor == nullptr;
> 
>       // For consistency with previous behavior, allow it even if it's been
>       // imported for some other property.
>       return (accessor && accessor->getAccessorKind() == accessorInfo->Kind);
>     }
> ```
> 
> ?

Hmm, on second thought I think this could be the case given:

```
      auto accessor = dyn_cast<AccessorDecl>(fn);
      if (!accessorInfo)
        return accessor == nullptr;

```